src/nih_wayfinding/test/spec/views/locations/locations-controller.spec.js

```diff
@@ -2,53 +2,87 @@
 
 describe('nih.views.locations: LocationsController', function () {
 
-  // load the controller's module
-  beforeEach(module('nih.views.locations'));
-  // load target module
-  beforeEach(module('nih.views.routing'));
-
-  var LocationsController;
-  var httpBackend;
-  var rootScope;
-  var scope;
-  var state;
-  var Geocoder;
-
-  // Initialize the controller and a mock scope
-  beforeEach(inject(function ($controller, $httpBackend, $q, $rootScope, $state) {
-    httpBackend = $httpBackend;
-    rootScope = $rootScope;
-    scope = $rootScope.$new();
-    state = $state;
-    Geocoder = {
-        search: function (searchText) {
-            var dfd = $q.defer();
-            dfd.resolve([{
-                geometry: {
-                    x: 10,
-                    y: 10
+    // load the controller's module
+    beforeEach(module('nih.views.locations'));
+    // load target module
+    beforeEach(module('nih.views.routing'));
+
+    var LocationsController;
+    var httpBackend;
+    var rootScope;
+    var scope;
+    var state;
+    var Geocoder;
+    var Notifications;
+
+    // Initialize the controller and a mock scope
+    beforeEach(inject(function ($httpBackend, $rootScope, $state, _Notifications_) {
+        httpBackend = $httpBackend;
+        rootScope = $rootScope;
+        scope = $rootScope.$new();
+        state = $state;
+        Notifications = _Notifications_;
```

Comment:
For angular components (anything prefixed with $), inject with the normal name and make the local version not prefixed with a $. This indicates that it is potentially a mocked service, and not the actual implementation.

For any other component, use the underscore notation and name the local version whatever its standard name is.

The underscore notation is a bit of angular injector magic where it strips the underscores. This allows you to maintain the original names of components in your tests. Otherwise var names would get weird, or you'd have to prefix all your local vars with something else, which is less readable.